zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/FileSystemStorage.java

```diff
@@ -30,9 +30,31 @@
 
   private static Logger LOGGER = LoggerFactory.getLogger(FileSystemStorage.class);
 
+  // only do UserGroupInformation.loginUserFromKeytab one time, otherwise you will still get
+  // your ticket expired.
+  static {
+    if (UserGroupInformation.isSecurityEnabled()) {
+      ZeppelinConfiguration zConf = ZeppelinConfiguration.create();
+      String keytab = zConf.getString(
+          ZeppelinConfiguration.ConfVars.ZEPPELIN_SERVER_KERBEROS_KEYTAB);
+      String principal = zConf.getString(
+          ZeppelinConfiguration.ConfVars.ZEPPELIN_SERVER_KERBEROS_PRINCIPAL);
+      if (StringUtils.isBlank(keytab) || StringUtils.isBlank(principal)) {
+        throw new RuntimeException("keytab and principal can not be empty, keytab: " + keytab
+            + ", principal: " + principal);
```

Comment:
is `static Logger LOGGER` created? could we log here?

Reply:
It would be logged by outside code  (caller of FileSystemStorage)